src/util/normalize-manifest/resolve-relative.js

```diff
@@ -0,0 +1,52 @@
+import {FILE_PROTOCOL_PREFIX} from '../../resolvers/exotics/file-resolver.js';
+import {LINK_PROTOCOL_PREFIX} from '../../resolvers/exotics/link-resolver.js';
+
+const path = require('path');
+
+export default function(info: Object, moduleLoc: string, lockfileFolder: string): void {
+  // It won't work if we don't yet know what's the folder we'll use as root. It's not a
+  // big deal tho, because it only happens when trying to figure out the root, and we
+  // don't need to know the dependencies / devDependencies at this time.
+  if (!lockfileFolder) {
+    return;
+  }
+
+  for (const hint of ['dependencies', 'devDependencies', 'optionalDependencies', 'peerDependencies']) {
```

Comment:
Can you move the `['dependencies', 'devDependencies', 'optionalDependencies', 'peerDependencies']` part into a constant so we don't have to create this array ever again in anywhere in the app?